src/app/modules/layout/components/admin-layout/admin-layout.component.ts

```diff
@@ -41,150 +38,95 @@ import { waitForSystemInfo } from 'app/store/system-info/system-info.selectors';
-        }
-      });
-    }
-
-    if (this.media.isActive('xs') || this.media.isActive('sm')) {
-      this.isSidenavOpen = false;
-    }
     this.sysGeneralService.toggleSentryInit();
-
     this.store$.pipe(waitForGeneralConfig, untilDestroyed(this)).subscribe((config) => {
       this.languageService.setLanguage(config.language);
     });
-
-    this.isSidenavCollapsed = this.layoutService.isMenuCollapsed;
-
     this.store$.dispatch(adminUiInitialized());
-
-    this.tokenLifetimeService.start();
+    this.listenToSidenavChanges();
   }
 
   ngOnDestroy(): void {
     this.tokenLifetimeService.stop();
   }
 
   ngAfterViewInit(): void {
+    this.sidenavService.setSidenav(this.sideNavs?.first);
     this.renderPageHeader();
   }
 
-  onSidenavStatusChange(sidenav: SidenavStatusData): void {
-    this.isSidenavOpen = sidenav.isOpen;
-    this.sidenavMode = sidenav.mode;
-    this.isSidenavCollapsed = sidenav.isCollapsed;
-  }
-
-  updateSidenav(): void {
-    this.isSidenavOpen = !this.isMobile;
-    this.sidenavMode = this.isMobile ? 'over' : 'side';
-    if (!this.isMobile) {
-      // TODO: This is hack to resolve issue described here: https://ixsystems.atlassian.net/browse/NAS-110404
-      setTimeout(() => {
-        this.sideNav.open();
-      });
-      this.store$.pipe(
-        waitForPreferences,
-        take(1),
-        filter((preferences) => Boolean(preferences.sidenavStatus)),
-        untilDestroyed(this),
-      ).subscribe(({ sidenavStatus }) => {
-        this.layoutService.isMenuCollapsed = sidenavStatus.isCollapsed;
-        this.isSidenavCollapsed = sidenavStatus.isCollapsed;
-      });
-    } else {
-      this.layoutService.isMenuCollapsed = false;
-    }
-    this.cdr.detectChanges();
-  }
-
-  get sidenavWidth(): string {
-    const iconified = this.layoutService.isMenuCollapsed;
-    if (this.isSidenavOpen && iconified && this.sidenavMode === 'side') {
-      return '48px';
-    }
-    if (this.isSidenavOpen && !iconified && this.sidenavMode === 'side') {
-      return '240px';
-    }
-    return '0px';
+  listenToSidenavChanges(): void {
+    this.sideNavs?.changes.pipe(untilDestroyed(this)).subscribe(() => {
+      this.sidenavService.setSidenav(this.sideNavs.first);
+    });
   }
 
   // For the slide-in menu
   toggleMenu(menuInfo?: [string, SubMenuItem[]]): void {
     const [state, subItems] = menuInfo || [];
-    if ((this.isOpen && !menuInfo) || (this.isOpen && state === this.menuName)) {
-      this.isOpen = false;
+    if ((this.isOpenSecondaryMenu && !menuInfo) || (this.isOpenSecondaryMenu && state === this.menuName)) {
+      this.isOpenSecondaryMenu = false;
       this.subs = [];
     } else if (menuInfo) {
       this.menuName = state;
       this.subs = subItems;
-      this.isOpen = true;
+      this.isOpenSecondaryMenu = true;
     }
   }
 
   onMenuClosed(): void {
-    this.isOpen = false;
+    this.isOpenSecondaryMenu = false;
```

Comment:
This `isOpenSecondaryMenu` should also be part of the nav service instead of completely being managed by the component